### ISSUE TITLE
Bad error message for FOR UPDATE on non-existent table

### ISSUE THREAD

**Issue by hlinnaka** [greenplum-db/gpdb#5915]
On GPDB master only (works on 5.x):
```
postgres=# select a from xx for update;
ERROR:  Got Invalid RelationId of xx (parse_relation.c:1027)
```

**Comment by BaiShaoqi** [comment]
Could I use the following code to deal with issue?  At least it can have the right output 
```
postgres=# select a from xx for update;
ERROR:  relation "xx" does not exist

```

```
diff --git a/src/backend/parser/parse_relation.c b/src/backend/parser/parse_relation.c
index c1b4163..bc23635 100644
--- a/src/backend/parser/parse_relation.c
+++ b/src/backend/parser/parse_relation.c
@@ -1055,7 +1055,7 @@ addRangeTableEntry(ParseState *pstate,
                {
                        Oid relid;

-                       relid = RangeVarGetRelid(relation, lockmode, true);
+                       relid = RangeVarGetRelid(relation, lockmode, false);
                        if (relid == InvalidOid)
                                elog(ERROR, "Got Invalid RelationId of %s", relation->relname);
```

**Comment by hlinnaka** [comment]
Yep. And once you do that, you can remove the check for InvalidOid.

**Comment by BaiShaoqi** [comment]
Yes, I will. I didn't realize that, thanks for remind me.

**Comment by hlinnaka** [comment]
This was fixed in commit 68f0d68324. Closing, thanks!
